src/__tests__/productlisting.test.js

```diff
@@ -0,0 +1,16 @@
+import { BrowserRouter as Router } from 'react-router-dom';
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+import ProductPage from '../components/ProductListing';
+import store from '../redux/configureStore';
+
+it('renders properly', () => {
+  const tree = renderer.create(
+    <Router>
+      <Provider store={store}>
+        <ProductPage />
+      </Provider>
+    </Router>,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```

Comment:
- [ ] Good job so far, Kindly perform a mocking data fetching and Redux Store connection. It is required that your app has integration tests using React Testing Library. You can check out this [link](https://testing-library.com/docs/react-testing-library/example-intro/) to learn how to do this.

Reply:
Created redux-react integration tests.